### ISSUE TITLE
[XPU] Support toggling profiler on/off for XPU.

### ISSUE THREAD

**Issue by frost-intel** [pytorch/pytorch#154898]
### 🚀 The feature, motivation and pitch

When attempting to toggle the profiler dynamically on XPU, this is not currently supported. Currently, this depends on:

* [x]  Kineto code change - [Add toggleCollectionDynamic for XPUPTI kineto#1088](https://github.com/pytorch/kineto/pull/1088)
* [ ]  Update Kineto pin in PyTorch
* [ ]  XPU code adaptation in PyTorch - #155135

Toggling the profiler is an option for preventing profiling during some sensitive/expensive block of code.
```python
import torch
from torch.profiler import profile, ProfilerActivity
with profile(activities=[ProfilerActivity.XPU]) as p:
    x = torch.ones(10).xpu()
    p.toggle_collection_dynamic(False, [ProfilerActivity.XPU])
    x1 = x * 2.5
    p.toggle_collection_dynamic(True, [ProfilerActivity.XPU])
    x / 4.5
```
Currently, the output is not correct. Output:
```
[W602 20:51:07.921016123 profiler_kineto.cpp:708] Dynamic toggle is only supported for CPU/GPU activity, skipping toggling of XPU
[W602 20:51:07.966075439 profiler_kineto.cpp:708] Dynamic toggle is only supported for CPU/GPU activity, skipping toggling of XPU
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  
                                                   Name    Self CPU %      Self CPU   CPU total %     CPU total  CPU time avg      Self XPU    Self XPU %     XPU total  XPU time avg    # of Calls  
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  
                                       urUSMDeviceAlloc        12.50%     213.972us        12.50%     213.972us     213.972us       0.000us         0.00%       0.000us       0.000us             1  
                                     urEnqueueUSMMemcpy        14.92%     255.385us        14.92%     255.385us     255.385us       0.000us         0.00%       0.000us       0.000us             1  
                 Memcpy M2D (MEMORY(Unknown) -> DEVICE)         0.00%       0.000us         0.00%       0.000us       0.000us       4.720us        22.61%       4.720us       4.720us             1  
                                  urEnqueueKernelLaunch        72.58%       1.242ms        72.58%       1.242ms     621.214us       0.000us         0.00%       0.000us       0.000us             2  
at::native::xpu::VectorizedElementwiseKernel<4, at::...         0.00%       0.000us         0.00%       0.000us       0.000us       7.840us        37.55%       7.840us       7.840us             1  
at::native::xpu::VectorizedElementwiseKernel<4, at::...         0.00%       0.000us         0.00%       0.000us       0.000us       8.320us        39.85%       8.320us       8.320us             1  
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  
Self CPU time total: 1.712ms
Self XPU time total: 20.880us
```
In practice, we would expect to only see *one* `xpu::VectorizedElementwiseKernel` but instead we see two, because the profiler is not actually toggling off.

Dynamically toggling the profiler can be helpful when only concerned with a specific section of code, especially in a distributed environment. Specifically, distributed XPU profiling in the scale-out (multi-node) setting is seeing some bugs and hangs that will not be addressed until Pytorch 2.9. In order to provide distributed profiling functionality in 2.8, we would need to support toggling XPU profiling on/off.

Adding this functionality requires both changes in kineto as well as Pytorch.

cc @robieta @chaekit @guotuofeng @guyang3532 @dzhulgakov @davidberard98 @briancoutinho @sraikund16 @sanrise @gujinghui @EikanWang @fengyuan14 @guangyey @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

**Comment by frost-intel** [comment]
Proposed change for kineto is at https://github.com/pytorch/kineto/pull/1088

@zejun-chen @gujinghui @guangyey

**Comment by malfet** [comment]
@frost-intel please do not use `intel` label on issue, it's intended only for PRs (and to be frank, unsure what is the purpose of this one, anyone knows?)

**Comment by EikanWang** [comment]
@frost-intel, could you please provide an informative description, including motivation and impact? And please share some real scenarios here.

**Comment by guangyey** [comment]
I think @frost-intel would like to support the following scenario for XPU
```python
import torch
from torch.profiler import profile, ProfilerActivity
with profile(activities=[ProfilerActivity.XPU]) as p:
    x = torch.ones(10).xpu()
    p.toggle_collection_dynamic(False, [ProfilerActivity.XPU])
    x1 = x * 2.5
    p.toggle_collection_dynamic(True, [ProfilerActivity.XPU])
    x / 4.5
```
Currently, it depends on

- [ ] Kineto code change - https://github.com/pytorch/kineto/pull/1088
- [ ] Update Kineto pin in PyTorch
- [ ] Maybe some XPU code adaptation in PyTorch

**Comment by zejun-chen** [comment]
Hi, @frost-intel 

You need the dynamic collection to toggle the profiling for fine-grained control which workloads to be traced down on intel GPUs right? Here is the standard usage of this feature: https://docs.pytorch.org/docs/stable/profiler.html#torch.profiler._KinetoProfile.toggle_collection_dynamic

I think your requirement makes sense. It will help frontend profiler users.
Thank you.
